resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/StringParameterInjector.java

```diff
@@ -80,40 +80,39 @@ protected void initialize(Class type, Type genericType, String paramName, Class
       baseType = type;
       baseGenericType = genericType;
 
-      if (type.isArray()) baseType = type.getComponentType();
-      if (List.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = ArrayList.class;
-      }
-      else if (SortedSet.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = TreeSet.class;
-      }
-      else if (Set.class.isAssignableFrom(type))
+      if (type.isArray())
       {
-         isCollection = true;
-         collectionType = HashSet.class;
+    	  baseType = type.getComponentType();
       }
-      if (isCollection)
+     
+      if (!baseType.isPrimitive())
       {
-         if (genericType != null && genericType instanceof ParameterizedType)
+         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
+         if (paramConverter != null) 
          {
-            ParameterizedType zType = (ParameterizedType) genericType;
-            baseType = Types.getRawType(zType.getActualTypeArguments()[0]);
-            baseGenericType = zType.getActualTypeArguments()[0];

```

Comment:
This is interesting. Really, I don't understand why baseGenericType is getting modified here. Given that there may have been some justification that I don't see, I appreciate that you added the call on line 90 as an addition to, rather than a replacement of, the original call.